main.tf

```diff
@@ -47,18 +47,28 @@ module "acm_certificate" {
   providers = { aws.certificate = aws.cluster, aws.route53 = aws.route53 }
   source    = "./modules/acm-certificate"
 
-  domain_name      = each.value
-  hosted_zone_name = var.validate_certificates ? var.hosted_zone_name : null
+  domain_name = each.value
+
+  hosted_zone_name = (
+    var.validate_certificates ?
+    try(var.additional_hosted_zones[each.value], var.hosted_zone_name) :
```

Comment:
Oh this is a neat keyword, I didn't realize terraform had `try`.

Reply:
Would it make sense to move this: `try(var.additional_hosted_zones[each.value], var.hosted_zone_name)` out into a local? Then we could refer to just the normalized value instead of needing to use try in multiple spots.